app/controllers/member_authentications_controller.rb

```diff
@@ -34,6 +34,11 @@ def member
     @member ||= Member.find_by(id: cookies.signed[:member_id], email: params[:email])
   end
 
+  def localize_by_recent_action
+    code = member&.actions&.sort_by(&:created_at)&.first&.page&.language&.code
```

Comment:
Did you rush this line? It's quite the train. Also, why `sort_by`, what's wrong with `last`:
```ruby
  member.actions.last.& etc
```